java/core/src/main/java/co/worklytics/psoxy/gateway/impl/CompositeConfigService.java

```diff
@@ -0,0 +1,47 @@
+package co.worklytics.psoxy.gateway.impl;
+
+import co.worklytics.psoxy.gateway.ConfigService;
+import lombok.AccessLevel;
+import lombok.Builder;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+
+import java.util.Optional;
+
+/**
+ * constructs a composite ConfigService built from two others
+ *
+ * checks 'preferred' first, and then fallback if not found there.
+ *
+ * although in practice we don't expect a given property to be defined in both, the value from
+ * 'preferred' would be taken in such a scenario
+ *
+ */
+@Builder
+@RequiredArgsConstructor(access = AccessLevel.PACKAGE)
+public class CompositeConfigService implements ConfigService {
```

Comment:
Not sure if this is a good idea. I get the point of simplifying the config access, but maybe the AWS implementation needs to be aware of what comes from global and what comes from lambda config to avoid misconfigurations (i.e. a global overwriting lambda, or f.e. the SALT not being in the global and lambdas with different SALTs? )